airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -310,7 +310,7 @@ def execute(self, context) -> Optional[str]:
                 self.log.info("creating pod with labels %s and launcher %s", labels, launcher)
                 final_state, _, result = self.create_new_pod_for_operator(labels, launcher)
             if final_state != State.SUCCESS:
-                status = self.client.read_namespaced_pod(self.name, self.namespace)
+                status = self.client.read_namespaced_pod(self.pod.metadata.name, self.namespace)
                 raise AirflowException(f'Pod returned a failure: {status}')
```

Comment:
```suggestion
                raise AirflowException(f'Pod {self.pod.metadata.name} returned a failure: {status}')
```

Lets include the name of the pod that failed too